packages/analytics/dashboard-renderer/src/types/dashboard-renderer-types.ts

```diff
@@ -61,42 +81,116 @@ export const gaugeChartSchema = {
 
 export type GaugeChartOptions = FromSchema<typeof gaugeChartSchema>
 
-export const tileSchema = {
+export const tileDefinitionSchema = {
   type: 'object',
   properties: {
     query: {
       // TODO: JSON Schema for Explore v4.
       type: 'object',
     },
     chart: {
-      oneOf: [barChartSchema, gaugeChartSchema],
+      oneOf: [barChartSchema, gaugeChartSchema, timeseriesChartSchema],
     },
-    title: {
+  },
+  required: ['query', 'chart'],
+  additionalProperties: false,
+} as const satisfies JSONSchema
+
+export type TileDefinition = FromSchema<typeof tileDefinitionSchema>
+
+export const tileLayoutSchema = {
+  type: 'object',
+  properties: {
+    position: {
+      type: 'object',
+      properties: {
+        col: {
+          type: 'number',
+        },
+        row: {
+          type: 'number',
+        },
+      },
+      description: 'Position of the tile in the grid.',
+      required: ['col', 'row'],
+      additionalProperties: false,
+    },
+    size: {
+      type: 'object',
+      properties: {
+        cols: {
+          type: 'number',
+        },
+        rows: {
+          type: 'number',
+        },
+      },
+      description: 'Number of columns and rows the tile occupies.',
+      required: ['cols', 'rows'],
+      additionalProperties: false,
+    },
+  },
+  required: ['position', 'size'],
+  additionalProperties: false,
+} as const satisfies JSONSchema
+
+export type TileLayout = FromSchema<typeof tileLayoutSchema>
+
+export const tileConfigSchema = {
+  type: 'object',
+  properties: {
+    id: {
       type: 'string',
     },
+    definition: {
+      allOf: [tileDefinitionSchema],
```

Comment:
You can probably simplify this by just embedding the `tileDefinitionSchema` directly: `definition: tileDefinitionSchema`.

Reply:
This doesn't work, FromSchema fails, results in `TileConfig` being type `never` 

![image](https://github.com/Kong/public-ui-components/assets/6026470/93bf637c-6e6b-45f3-b979-d9ee936e9f20)

Reply:
never mind